### PR TITLE
Updating tests and POST UserActiveGames endpoint is functional

### PR DESCRIPTION
**Pull request by DarthofLuke**
Refactored "InProgressGames" to "ActiveGames"
Restructured userActiveGames schema based on feedback
Changed collection name from user_game_results to user_game_info
Adding routing and validation for UserActiveGames endpoint.

### CODE REVIEW

File: src/app/controllers/userActiveGames.controller.ts [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+ * @param next This takes us to the errorHandler if request fails
+ */
+async function createUserActiveGame(req, res, next) {
+
+    const allData = Object.values(matchedData(req, { locations: ['body'] }));
+    console.log(allData.values());
+    // this is needed because the last element in matchedData array is the original request for some reason.
+    // I believe this is a bug with express-validator relating to bodies that have arrays
+    console.log(allData);
+    console.log("Hello!" + allData);
```
Comment (Gregory711) [inline src/app/controllers/userActiveGames.controller.ts:29]: allData is printed twice

Comment (DarthofLuke) [reply src/app/controllers/userActiveGames.controller.ts]: Thanks for catching that, that is old debug code.

File: src/app/controllers/userActiveGames.controller.ts [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+ * Returns 201 if userActiveGameService is successful
+ * Otherwise catches error and sends to our errorHandler
+ * Takes sanitized input and sends it to userActiveGamesService
+ * @param req This is the request object
+ * @param res This is the response object
+ * @param next This takes us to the errorHandler if request fails
+ */
+async function createUserActiveGame(req, res, next) {
+
+    const allData = Object.values(matchedData(req, { locations: ['body'] }));
```
Comment (Gregory711) [inline src/app/controllers/userActiveGames.controller.ts:24]: Why not just set allData to req.body?

Comment (DarthofLuke) [reply src/app/controllers/userActiveGames.controller.ts:24]: I'm following the documentation for express-validator: https://express-validator.github.io/docs/matched-data-api

**Review by Gregory711 — APPROVED** [review]

**Pull request merged**